src/Examine.Lucene.Spatial/Search/Spatial4nShapeFactory.cs

```diff
@@ -0,0 +1,92 @@
+using System.Collections.Generic;
+using System.Linq;
+using Examine.Search;
+using Spatial4n.Context;
+using Spatial4n.Distance;
+using Spatial4n.Shapes;
+
+namespace Examine.Lucene.Spatial.Search
+{
+    public class Spatial4nShapeFactory : IExamineSpatialShapeFactory
+    {
+        private SpatialContext _spatialContext;
+        /// <summary>
+        /// Contructor
+        /// </summary>
+        /// <param name="spatialContext">Spatial Context</param>
+        public Spatial4nShapeFactory(SpatialContext spatialContext)
+        {
+            _spatialContext = spatialContext;
+        }
+
+        public SpatialContext SpatialContext { get => _spatialContext; }
+
+        /// <inheritdoc/>
+        public IExamineSpatialCircle CreateCircle(double x, double y, double distance)
+        {
+            var spatial4NCircle = _spatialContext.MakeCircle(x, y, distance);
+            return new ExamineLuceneCircle(spatial4NCircle);
+        }
+
+        /// <inheritdoc/>
+        public IExamineSpatialCircle CreateEarthEquatorialSearchRadiusKMCircle(double x, double y, double radius)
+        {
+            var spatial4NCircle = _spatialContext.MakeCircle(x, y, DistanceUtils.Dist2Degrees(radius, DistanceUtils.EarthEquatorialRadiusKilometers));
+            return new ExamineLuceneCircle(spatial4NCircle);
+        }
+
+        /// <inheritdoc/>
+        public IExamineSpatialCircle CreateEarthMeanSearchRadiusKMCircle(double x, double y, double radius)
+        {
+            var spatial4NCircle = _spatialContext.MakeCircle(x, y, DistanceUtils.Dist2Degrees(radius, DistanceUtils.EarthMeanRadiusKilometers));
+            return new ExamineLuceneCircle(spatial4NCircle);
+        }
+
+        /// <inheritdoc/>
+        public IExamineSpatialEmptyShape CreateEmpty()
+        {
+            return new ExamineLuceneEmptyShape();
+        }
+
+        /// <inheritdoc/>
+        public IExamineSpatialPoint CreateGeoPoint(double lattitude, double longitude)
+        {
+            //Swapped on purpose
+            double y = lattitude;
```

Comment:
@nzdev seems like a typo in latitude. Should be `latitude` instead of `lattitude`.

Reply:
Good catch